Lib/ufo2ft/filters/propagateAnchors.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from fontTools.misc.transform import Transform
+from ufo2ft.filters import BaseFilter
+
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class PropagateAnchorsFilter(BaseFilter):
+
+    def set_context(self, font, glyphSet):
+        ctx = super(PropagateAnchorsFilter, self).set_context(font, glyphSet)
+        ctx.processed = set()
+        return ctx
+
+    def __call__(self, font, glyphSet=None):
+        if super(PropagateAnchorsFilter, self).__call__(font, glyphSet):
+            modified = self.context.modified
+            if modified:
+                logger.info('Glyphs with propagated anchors: %i' %
+                            len(modified))
+            return modified
+
+    def filter(self, glyph):
+        if not glyph.components:
+            return False
+        _propagate_glyph_anchors(self.context.font, glyph,
```

Comment:
sorry actually this should be `self.context.glyphSet` (a dictionary of parent-less glyphs that are copies of the original glyphs in the font).
The general rule for filters like these is: you use `self.context.font` when you need to read (and only read) to global data that's in a font (lib, fontinfo, features, etc), whereas you use `self.context.glyphSet` when you need to reference other glyphs by name (which is your case here). The glyphs in the glyphSet are the ones currently being modified by this and all previous filters. The glyphs in the original font should be left untouched.

Reply:
Sure. It isn’t being modified in any case.